pkg/kubectl/cmd/proxy.go

```diff
@@ -107,6 +111,17 @@ func RunProxy(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
 		return err
 	}
 
-	glog.Fatal(server.Serve(port))
+	// Separate listening from serving so we can report the bound port
+	// when it is chosen by os (port == 0)
+	l, err := server.Listen(port)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	_, boundPort, err := net.SplitHostPort(l.Addr().String())
+	if err != nil {
+		glog.Fatal(err)
+	}
+	fmt.Fprintf(out, "Starting to serve on localhost:%s", boundPort)
```

Comment:
Do you want to indicate "localhost" or do you want to indicate the address you're actually bound to?  If I'm reading right, that would be all of them.